app/src/main/java/com/googleresearch/capturesync/SoftwareSyncController.java

```diff
@@ -275,20 +344,68 @@ public void close() {
         }
     }
 
-    private String lastFourSerial() {
-        String serial = Secure.getString(context.getContentResolver(), Secure.ANDROID_ID);
-        if (serial.length() <= 4) {
-            return serial;
+    /**
+     * Change the lock status of the settings selected for synchronization. If the settings are
+     * getting locked then the provided settings are saved for broadcasts.
+     *
+     * @param settings the locked settings to be saved.
+     * @throws IllegalArgumentException if null is provided when settings are to be locked.
+     */
+    public void switchSettingsLock(SyncSettingsContainer settings) {
+        if (isSettingsBroadcasting()) {
+            ((SoftwareSyncLeader) softwareSync).setSavedSettings(null);
+        } else if (settings != null) {
+            ((SoftwareSyncLeader) softwareSync).setSavedSettings(settings);
         } else {
-            return serial.substring(serial.length() - 4);
+            throw new IllegalArgumentException("Settings to be locked cannot be null");
         }
     }
 
+    /**
+     * Indicates whether this device is a leader.
+     *
+     * @return true if this device is a leader, false if it is a client.
+     */
     public boolean isLeader() {
         return isLeader;
     }
 
-    public TextView getStatusText() {
-        return statusView;
+    /**
+     * Indicates whether the last finished alignment attempt was successful.
+     *
+     * @return true if the last alignment attempt was successful, false if it wasn't or no attempts
+     * were made.
+     */
+    public boolean isAligned() {
+        return phaseAlignController.wasAligned();
+    }
+
+    /**
+     * Current broadcasting status of the leader's settings. If this is true then each new client
+     * will receive the previously saved settings of the leader.
+     *
+     * @return true if the settings are being broadcast, false if they are not.
+     * @throws IllegalStateException if the device if not a leader.
+     */
+    public boolean isSettingsBroadcasting() {
+        if (isLeader) {
+            return ((SoftwareSyncLeader) softwareSync).getSavedSettings() != null;
+        }
+        throw new IllegalStateException("Cannot check the settings lock status for a client");
+    }
+
+    /**
+     * Provides the given timestamp to {@link PeriodCalculator} and a converted to leader time
+     * domain version of the given timestamp to {@link PhaseAlignController}.
+     *
+     * @param timestamp a timestamp to be provided.
+     */
+    public void updateTimestamp(long timestamp) {
+        periodCalculator.onFrameTimestamp(timestamp);
```

Comment:
Could we setup different states for SoftwareSyncController and switch them during pre-recording/recording? Like, "period calculation", "phase alignment", "recording", etc. 
For example, here we don't need to process timestamps in period, if we are on the step of phase alignment and vice versa.

Reply:
I left updates for `phaseAlignController` even during period calculation because otherwise phase error text freezes on the preview